### PR TITLE
add Juneteenth to USFederalHolidayCalendar

### PR DESCRIPTION
**Pull request by chx5**
* Add **Juneteenth National Independence Day**

*  Standardize the [official federal holiday names](https://www.opm.gov/policy-data-oversight/pay-leave/federal-holidays/#url=2021)

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Can you write a test for this.

File: pandas/tseries/holiday.py [hunk 1]
```diff
@@ -559,6 +559,7 @@ class USFederalHolidayCalendar(AbstractHolidayCalendar):
         USMartinLutherKingJr,
         USPresidentsDay,
         USMemorialDay,
+        Holiday("Juneteenth", month=6, day=19, observance=nearest_workday),
```
Comment (twoertwein) [inline pandas/tseries/holiday.py:562]: Should probably be `"Juneteenth National Independence Day"` https://www.opm.gov/policy-data-oversight/pay-leave/federal-holidays/#url=2021

**Comment by datapythonista** [comment]
Thanks @ChuliangXiao, besides the test, can you also please an entry in the release notes? Thanks!

**Comment by datapythonista** [comment]
Closing as stale, but if you want to finish this @ChuliangXiao just let me know.

**Pull request closed without merging**